src/SonarQube.Bootstrapper/SonarQube.Bootstrapper.csproj

```diff
@@ -18,4 +18,15 @@
     <ProjectReference Include="..\SonarQube.TeamBuild.PreProcessor\SonarQube.TeamBuild.PreProcessor.csproj" />
     <ProjectReference Include="..\SonarScanner.Shim\SonarScanner.Shim.csproj" />
   </ItemGroup>
-</Project>
+  <!-- Conditionally reference project only if the framework is classic .NET Framework (not core or standard). -->
+  <ItemGroup Condition="('$(TargetFrameworkIdentifier)' == '.NETFramework')">
+      <ProjectReference Include="..\SonarQube.TeamBuild.Integration.Legacy\SonarQube.TeamBuild.Integration.Legacy.csproj" />
+  </ItemGroup>
+  <!-- Define a constant if the framework is classic .NET Framework. -->
+  <Target Name="AddNetFrameworkMacro" BeforeTargets="CoreCompile" Condition="('$(TargetFrameworkIdentifier)' == '.NETFramework')">
```

Comment:
This is useless there are already a lot of constants for net fwk version (see [microsoft tutorial](https://docs.microsoft.com/en-us/dotnet/core/tutorials/libraries)).

Reply:
This is cleaner than saying "is it net45? or 46? 471? 583?". We don't care about the version, only the type of the framework. When we change target version in the future, we wouldn't need to modify that condition.

Reply:
I don't know which of the 2 I do prefer so let's say ok.